src/main/java/org/mskcc/cbio/oncokb/service/SlackService.java

```diff
@@ -263,24 +265,21 @@ public List<LayoutBlock> buildBlocks(UserDTO userDTO, boolean trialAccountActiva
 
     private LayoutBlock buildCollapsedBlock(UserDTO userDTO, ActionId actionId) {
         StringBuilder sb = new StringBuilder();
-        sb.append(userDTO.getEmail() + "\n" + userDTO.getCompanyName() + " (" + userDTO.getLicenseType().getShortName() + (withTrialAccountNote(userDTO, actionId) ? ", *TRIAL*)" : userDTO.isActivated() ? ")" : ")\n*NOT ACTIVATED*: "));
-        if (!userDTO.isActivated() && !withTrialAccountNote(userDTO, actionId)) {
-            if (withRejectionNote(userDTO, actionId)) {
-                sb.append("Sent rejection email");
-            } else if (withRejectAlumniAddressNote(userDTO, actionId)) {
-                sb.append("Rejected user due to alumni email address");
-            } else if (withUseCaseClarificationNote(userDTO, actionId)) {
-                sb.append("Sent use case clarification");
-            } else if (withForProfitClarificationNote(userDTO, actionId)) {
-                sb.append("Clarified with user on for-profit affiliation");
-            } else if (withAcademicClarificationNote(userDTO, actionId)) {
-                sb.append("Clarified with user on noninstitutional email");
-            } else if (withDuplicateUserClarificationNote(userDTO, actionId)) {
-                sb.append("Clarified with user on multiple account request");
-            } else if (withRegistrationInfoClarificationNote(userDTO, actionId)) {
-                sb.append("Clarified with user on registration info");
-            } else if (withLicenseOptionsNote(userDTO, actionId)) {
-                sb.append("Sent license options email");
+        sb.append(userDTO.getEmail() + "\n" + userDTO.getCompanyName() + " (" + userDTO.getLicenseType().getShortName() + (withNote(MailType.ACTIVATE_FREE_TRIAL, userDTO, actionId) ? ", *TRIAL*)" : userDTO.isActivated() ? ")" : ")\n*NOT ACTIVATED*: "));
+        if (!userDTO.isActivated() && !withNote(MailType.ACTIVATE_FREE_TRIAL, userDTO, actionId)) {
+            List<MailType> sentMails = new ArrayList<>();
+            for (MailType mailType : MailType.values()) {
+                if (withNote(mailType, userDTO, actionId))
+                    sentMails.add(mailType);
+            }
+            if (!sentMails.isEmpty()) {
+                sentMails = sentMails.stream().sorted((mt1, mt2) -> mt1.getCategory().orElse(null) == EmailCategory.DENY ? 1 : 0).collect(Collectors.toList());
```

Comment:
sort the rejection emails at the end, clarification/license emails before to have some sort of chronological nature still in the collapsed version.